src/jit/codegencommon.cpp

```diff
@@ -7595,20 +7598,22 @@ void CodeGen::genProfilingLeaveCallback(unsigned helper /*= CORINFO_HELP_PROF_FC
         if (RBM_ARG_0 & gcInfo.gcRegGCrefSetCur)
         {
             attr = EA_GCREF;
-            gcInfo.gcMarkRegSetGCref(RBM_PROFILER_RET_SCRATCH);
+            gcInfo.gcMarkRegSetGCref(RBM_PROFILER_RET_SCRATCH1 | RBM_PROFILER_RET_SCRATCH2);
         }
         else if (RBM_ARG_0 & gcInfo.gcRegByrefSetCur)
         {
             attr = EA_BYREF;
-            gcInfo.gcMarkRegSetByref(RBM_PROFILER_RET_SCRATCH);
+            gcInfo.gcMarkRegSetByref(RBM_PROFILER_RET_SCRATCH1 | RBM_PROFILER_RET_SCRATCH2);
         }
         else
         {
             attr = EA_4BYTE;
         }
 
-        getEmitter()->emitIns_R_R(INS_mov, attr, REG_PROFILER_RET_SCRATCH, REG_ARG_0);
-        regTracker.rsTrackRegTrash(REG_PROFILER_RET_SCRATCH);
+        getEmitter()->emitIns_R_R(INS_mov, attr, REG_PROFILER_RET_SCRATCH1, REG_ARG_0);
+        getEmitter()->emitIns_R_R(INS_mov, attr, REG_PROFILER_RET_SCRATCH2, REG_ARG_1);
```

Comment:
It is desirable that the amount of jitted code needed to make an enter/leave/tailcall callback is as small as possible.  With the change of contract, now we will be generating one additional move for leave callback compared to earlier.  Though not a big deal, this could be avoided if we don't change the contract.